packages/insomnia-app/app/plugins/index.js

```diff
@@ -55,6 +55,12 @@ export type WorkspaceAction = {
   icon?: string,
 };
 
+export type PluginTab = {
+  plugin: Plugin,
+  label: string,
+  icon?: string,
+};
```

Comment:
I think you'll need an additional property here to return a raw DOM node?

```suggestion
  panelBody: (...) => HTMLElement | Promise<HTMLElement>,
};
```

Note, there is a `HtmlElementWrapper` component to place the DOM node returned onto the React tree.

Reply:
@codeasashu this `body` prop needs to be a function (so that context can be provided to a plugin). See my suggestion above.

Reply:
Fixed in 484781445ef509c090ee549e440efb1865c6dd54